prov/rxm/src/rxm_conn.c

```diff
@@ -350,6 +338,43 @@ static int rxm_conn_handle_notify(struct fi_eq_entry *eq_entry)
 	}
 }
 
+static void rxm_conn_handle_close(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn)
+{
+	struct util_cmap_cmd *cmd;
+	struct rxm_cmap_cmd_data *cmd_data;
+
+	cmd = calloc(1, sizeof(struct util_cmap_cmd) +
+			sizeof(struct rxm_cmap_cmd_data));
+	if (!cmd) {
+		free(rxm_conn->close_data);
+		rxm_conn_close_msg_ep(rxm_ep, rxm_conn);
+		return;
+	}
+	cmd->type = UTIL_CMAP_CMD_CONN_CLOSE;
+	cmd_data = (struct rxm_cmap_cmd_data *)cmd->data;
+	cmd_data->rxm_conn = rxm_conn;
+	cmd_data->close_cmd.close_data = rxm_conn->close_data;
+
+	rxm_conn->close_data = NULL;
+
+	dlist_ts_insert_tail(&rxm_ep->util_ep.cmap->cmd_queue, &cmd->entry);
+	rxm_ep->util_ep.cmap->cmd_write++;
+	rxm_conn_wake_up_wait_obj(rxm_ep);
+}
+
+static void rxm_conn_handle_conn_refused(struct rxm_ep *rxm_ep, struct rxm_conn *rxm_conn)
+{
+	if (rxm_conn->close_data) {
+		FI_DBG(&rxm_prov, FI_LOG_FABRIC,
+		       "Enqueue CONN_CLOSE req to CMD queue\n");
+		assert(rxm_conn->handle.state != CMAP_CONNREQ_SENT);
+		rxm_conn_handle_close(rxm_ep, rxm_conn);
```

Comment:
It's not clear how rxm_conn->close_data would be set at this point.

Reply:
rxm_conn->close_data is allocated when handling simultaneous connection request from passive side of connection (that has lower address). When ECONNREFUSED error received that notifies us about REJECT event, we will close MSG EP that contains in close_data.